api/src/main/java/org/allaymc/api/blockentity/component/BlockEntityFlowerPotBaseComponent.java

```diff
@@ -0,0 +1,65 @@
+package org.allaymc.api.blockentity.component;
+
+import org.allaymc.api.block.type.BlockState;
+import org.allaymc.api.item.ItemStack;
+
+/**
+ * @author Cdm2883
+ */
+public interface BlockEntityFlowerPotBaseComponent extends BlockEntityBaseComponent {
+
+    /**
+     * Get the plant inside the flower pot.
+     *
+     * @return the plant block state.
+     */
+    BlockState getPlantBlock();
+
+    /**
+     * Set the plant inside the flower pot.
+     *
+     * @param block the plant block state.
+     *
+     * @return {@code true} if it set successfully, otherwise {@code false}.
+     */
+    boolean setPlantBlock(BlockState block);
+
+    /**
+     * Get the plant as item inside the flower pot.
+     *
+     * @return the plant item.
```

Comment:
Describe this return value in detail (e.g. `@return the plant item, or {@code null} if the plant is empty.`)